confgen: potential deadlock problem

See testing failure: https://github.com/tableauio/tableau/actions/runs/16586300090/job/46912179956
```
panic: test timed out after 30m0s
running tests:
	TestGenerator_GenWorkbook (30m0s)
	TestGenerator_GenWorkbook/test1-FirstPassModeDefault (30m0s)

goroutine 8 [running]:
testing.(*M).startAlarm.func1()
	/opt/hostedtoolcache/go/1.21.13/x64/src/testing/testing.go:2259 +0x259
created by time.goFunc
	/opt/hostedtoolcache/go/1.21.13/x64/src/time/sleep.go:176 +0x45

goroutine 1 [chan receive, 30 minutes]:
testing.(*T).Run(0xc0004831e0, {0x152a64c, 0x19}, 0x1578b70)
	/opt/hostedtoolcache/go/1.21.13/x64/src/testing/testing.go:1649 +0x871
testing.runTests.func1(0x0?)
	/opt/hostedtoolcache/go/1.21.13/x64/src/testing/testing.go:2054 +0x85
testing.tRunner(0xc0004831e0, 0xc0004779e8)
	/opt/hostedtoolcache/go/1.21.13/x64/src/testing/testing.go:1595 +0x262
testing.runTests(0xc0001d59a0?, {0x1f3ade0, 0x1d, 0x1d}, {0x7fb6165c0108?, 0xc000477af8?, 0x1f91060?})
	/opt/hostedtoolcache/go/1.21.13/x64/src/testing/testing.go:2052 +0x8ae
testing.(*M).Run(0xc0001d59a0)
	/opt/hostedtoolcache/go/1.21.13/x64/src/testing/testing.go:1925 +0xcd8
github.com/tableauio/tableau/internal/protogen.TestMain(0x1510b64?)
	/home/runner/work/tableau/tableau/internal/protogen/protogen_test.go:33 +0x26f
main.main()
	_testmain.go:139 +0x328

goroutine 88 [chan receive, 30 minutes]:
testing.(*T).Run(0xc0002e56c0, {0x152bd65, 0x1a}, 0xc0001bdcd0)
	/opt/hostedtoolcache/go/1.21.13/x64/src/testing/testing.go:1649 +0x871
github.com/tableauio/tableau/internal/protogen.TestGenerator_GenWorkbook(0x0?)
	/home/runner/work/tableau/tableau/internal/protogen/protogen_test.go:167 +0x15af
testing.tRunner(0xc0002e56c0, 0x1578b70)
	/opt/hostedtoolcache/go/1.21.13/x64/src/testing/testing.go:1595 +0x262
created by testing.(*T).Run in goroutine 1
	/opt/hostedtoolcache/go/1.21.13/x64/src/testing/testing.go:1648 +0x846

goroutine 89 [semacquire, 30 minutes]:
sync.runtime_Semacquire(0xc000629d90?)
	/opt/hostedtoolcache/go/1.21.13/x64/src/runtime/sema.go:62 +0x25
sync.(*WaitGroup).Wait(0xc000629d88)
	/opt/hostedtoolcache/go/1.21.13/x64/src/sync/waitgroup.go:116 +0xa5
golang.org/x/sync/errgroup.(*Group).Wait(0xc000629d80)
	/home/runner/go/pkg/mod/golang.org/x/sync@v0.6.0/errgroup/errgroup.go:56 +0x38
github.com/tableauio/tableau/internal/protogen.(*Generator).processWorkbookOnFirstPass(0xc00037b710, {0xc0003f88e0, 0x2, 0xc000439b30?})
	/home/runner/work/tableau/tableau/internal/protogen/protogen.go:178 +0x2e5
github.com/tableauio/tableau/internal/protogen.(*Generator).GenWorkbook(0xc00037b710, {0xc0003f88e0?, 0x2, 0x2})
	/home/runner/work/tableau/tableau/internal/protogen/protogen.go:155 +0x545
github.com/tableauio/tableau/internal/protogen.TestGenerator_GenWorkbook.func1(0xc0002e5860)
	/home/runner/work/tableau/tableau/internal/protogen/protogen_test.go:168 +0x85
testing.tRunner(0xc0002e5860, 0xc0001bdcd0)
	/opt/hostedtoolcache/go/1.21.13/x64/src/testing/testing.go:1595 +0x262
created by testing.(*T).Run in goroutine 88
	/opt/hostedtoolcache/go/1.21.13/x64/src/testing/testing.go:1648 +0x846

goroutine 94 [sync.RWMutex.RLock, 30 minutes]:
sync.runtime_SemacquireRWMutexR(0xc00062ff90?, 0x20?, 0x4ad2b9?)
	/opt/hostedtoolcache/go/1.21.13/x64/src/runtime/sema.go:82 +0x25
sync.(*RWMutex).RLock(0xc00062ff80)
	/opt/hostedtoolcache/go/1.21.13/x64/src/sync/rwmutex.go:71 +0x65
github.com/tableauio/tableau/internal/x/xproto.(*TypeInfos).GetByFullName(0xc00062ff80, {0xc00058b3a8, 0x13})
	/home/runner/work/tableau/tableau/internal/x/xproto/build.go:148 +0x57
github.com/tableauio/tableau/internal/x/xproto.(*TypeInfos).Get(0xc00062ff80, {0xc0001c6d11, 0xa})
	/home/runner/work/tableau/tableau/internal/x/xproto/build.go:143 +0x139
github.com/tableauio/tableau/internal/protogen.parseTypeDescriptor(0xc0001c6d04?, {0xc0001c6d0c, 0x10})
	/home/runner/work/tableau/tableau/internal/protogen/parser.go:138 +0x151
github.com/tableauio/tableau/internal/protogen.(*documentParser).parseMapField(0xc00058bcf0, 0xc000000180, 0xc000036660)
	/home/runner/work/tableau/tableau/internal/protogen/document_parser.go:86 +0x676
github.com/tableauio/tableau/internal/protogen.(*documentParser).parseField(0xc00058bcf0, 0x2?, 0xc000036660)
	/home/runner/work/tableau/tableau/internal/protogen/document_parser.go:37 +0x2b7
github.com/tableauio/tableau/internal/protogen.(*Generator).convertDocument(0xc00037b710, {0xc00065a810, 0xd}, {0xc00065a81e, 0x9}, 0x58?, 0x0)
	/home/runner/work/tableau/tableau/internal/protogen/protogen.go:367 +0x1aeb
github.com/tableauio/tableau/internal/protogen.(*Generator).convertWithErrorModule(0xc00065a810?, {0xc00065a810, 0xd}, {0xc00065a81e, 0x9}, 0xa0?, 0x674b71?)
	/home/runner/work/tableau/tableau/internal/protogen/protogen.go:305 +0x292
github.com/tableauio/tableau/internal/protogen.(*Generator).processWorkbookOnFirstPass.func1()
	/home/runner/work/tableau/tableau/internal/protogen/protogen.go:175 +0x90
golang.org/x/sync/errgroup.(*Group).Go.func1()
	/home/runner/go/pkg/mod/golang.org/x/sync@v0.6.0/errgroup/errgroup.go:78 +0x98
created by golang.org/x/sync/errgroup.(*Group).Go in goroutine 89
	/home/runner/go/pkg/mod/golang.org/x/sync@v0.6.0/errgroup/errgroup.go:75 +0x125

goroutine 95 [sync.RWMutex.Lock, 30 minutes]:
sync.runtime_SemacquireRWMutex(0xc00062ff94?, 0x0?, 0xc0000a0a00?)
	/opt/hostedtoolcache/go/1.21.13/x64/src/runtime/sema.go:87 +0x25
sync.(*RWMutex).Lock(0xc00062ff80)
	/opt/hostedtoolcache/go/1.21.13/x64/src/sync/rwmutex.go:152 +0x8b
github.com/tableauio/tableau/internal/x/xproto.(*TypeInfos).Put(0xc00062ff80, 0xc00063a180)
	/home/runner/work/tableau/tableau/internal/x/xproto/build.go:122 +0x46
github.com/tableauio/tableau/internal/protogen.(*Generator).extractTypeInfoFromSpecialSheetMode(0xc00037b710, 0x2, 0xc0001440f0, {0xc0001ffd56, 0x4}, {0xc0001c6040, 0x1d})
	/home/runner/work/tableau/tableau/internal/protogen/protogen.go:567 +0x23a5
github.com/tableauio/tableau/internal/protogen.(*Generator).convertTable(0xc00037b710, {0xc0001ffce0, 0xc}, {0xc0001ffced, 0xe}, 0x58?, 0x0)
	/home/runner/work/tableau/tableau/internal/protogen/protogen.go:481 +0x1c9b
github.com/tableauio/tableau/internal/protogen.(*Generator).convertWithErrorModule(0xc0001ffce0?, {0xc0001ffce0, 0xc}, {0xc0001ffced, 0xe}, 0xf8?, 0x0?)
	/home/runner/work/tableau/tableau/internal/protogen/protogen.go:310 +0x3e5
github.com/tableauio/tableau/internal/protogen.(*Generator).processWorkbookOnFirstPass.func1()
	/home/runner/work/tableau/tableau/internal/protogen/protogen.go:175 +0x90
golang.org/x/sync/errgroup.(*Group).Go.func1()
	/home/runner/go/pkg/mod/golang.org/x/sync@v0.6.0/errgroup/errgroup.go:78 +0x98
created by golang.org/x/sync/errgroup.(*Group).Go in goroutine 89
	/home/runner/go/pkg/mod/golang.org/x/sync@v0.6.0/errgroup/errgroup.go:75 +0x125
FAIL	github.com/tableauio/tableau/internal/protogen	1800.133s
```

See full testing logs [job-logs.txt](https://github.com/user-attachments/files/21482369/job-logs.txt)

--- a/internal/x/xproto/build.go
+++ b/internal/x/xproto/build.go
@@ -111,6 +111,7 @@ func NewTypeInfos(protoPackage string) *TypeInfos {
 	}
 }
 
+// TypeInfos holds all type infos by full name mapping.
 type TypeInfos struct {
 	mu           sync.RWMutex
 	protoPackage string
@@ -121,7 +122,7 @@ type TypeInfos struct {
 func (x *TypeInfos) Put(info *TypeInfo) {
 	x.mu.Lock()
 	defer x.mu.Unlock()
-	log.Debugf("remember new generated predefined type: %v", info)
+	log.Debugf("add new type: %v", info)
 	x.infos[info.FullName] = info
 }
 
@@ -131,8 +132,6 @@ func (x *TypeInfos) Put(info *TypeInfo) {
 // prepended to generate full name. For example: ".ItemType" will be conveted to
 // "<ProtoPackage>.ItemType"
 func (x *TypeInfos) Get(name string) *TypeInfo {
-	x.mu.RLock()
-	defer x.mu.RUnlock()
 	var fullName string
 	if strings.HasPrefix(name, ".") {
 		// prepend default proto package

